File: oa/models/access.go
//
// access.go
// Copyright (C) 2024 veypi <i@veypi.com>
// 2024-09-23 16:21
// Distributed under terms of the MIT license.
//

package models

type Access struct {
	BaseDate
	AppID string `json:"app_id" gorm:"index;type:varchar(32)" methods:"post,list" parse:"json"`
	App   *App   `json:"-" gorm:"foreignKey:AppID;references:ID"`

	UserID *string `json:"user_id" gorm:"index;type:varchar(32);default: null" methods:"post,list" parse:"json"`
	User   *User   `json:"-" gorm:"foreignKey:UserID;references:ID"`

	RoleID *string `json:"role_id" gorm:"index;type:varchar(32);default: null" methods:"post,list" parse:"json"`
	Role   *Role   `json:"-" gorm:"foreignKey:RoleID;references:ID"`

	Name string `json:"name" methods:"post,*list" parse:"json"`

	TID   string `json:"tid" methods:"post" parse:"json"`
	Level uint   `json:"level" methods:"post" parse:"json"`
}

File: oaweb/composables/api/access.ts
//
// Copyright (C) 2024 veypi <i@veypi.com>
// 2024-10-11 14:36:07
// Distributed under terms of the MIT license.
//

import webapi from "./webapi"
import * as models from "./models"
export interface ListOpts {
  app_id: string
  user_id?: string
  role_id?: string
  name?: string
}
export interface ListQuery {
  created_at?: Date
  updated_at?: Date
}
export function List(json: ListOpts, query: ListQuery) {
  return webapi.Get<[models.Access]>(`/access`, { json, query })
}

export interface PostOpts {
  app_id: string
  user_id?: string
  role_id?: string
  name: string
  t_id: string
  level: number
}
export function Post(json: PostOpts) {
  return webapi.Post<models.Access>(`/access`, { json })
}

